automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainGroupRoleSpec.scala

```diff
@@ -0,0 +1,220 @@
+            AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(student.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ group member, + project owner, - group member" in {
+
+      // a billing project owner, a authdomain group owner and workspace owner
+      val billingProjectOwner = UserPool.chooseProjectOwner
+      implicit val token: AuthToken = billingProjectOwner.makeAuthToken()
+
+      // a billing project owner and a member of authdomain group
+      val student = UserPool.chooseStudent
+
+      withCleanBillingProject(billingProjectOwner) { projectName =>
+        withGroup("group", memberEmails = List(student.email)) { groupName =>
+          withCleanUp {
+            withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+
+              AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(student.makeAuthToken())
+
+              // add "student" to billing project with owner role
+              Rawls.billing.addUserToBillingProject(projectName, student.email, BillingProjectRole.Owner)
+              register cleanUp Try(Rawls.billing.removeUserFromBillingProject(projectName, student.email, BillingProjectRole.Owner)).recover {
+                case _: RestException =>
+              }
+              AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupName))(student.makeAuthToken())
+
+              // remove a group member "student" from group
+              Sam.user.removeUserFromPolicy(groupName, GroupRole.Member.toString, student.email)
+              AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(student.makeAuthToken())
+            }
+          }
+        }
+      }
+    }
+
+    "+ group member, + project owner, - project owner" in {
+
+      // a billing project owner, a authdomain group owner and workspace owner
+      val billingProjectOwner = UserPool.chooseProjectOwner
+      implicit val token: AuthToken = billingProjectOwner.makeAuthToken()
+
+      // a billing project owner and a member of authdomain group
+      val student = UserPool.chooseStudent
+
+      withCleanBillingProject(billingProjectOwner) { projectName =>
+        withGroup("group", memberEmails = List(student.email)) { groupName =>
+          withCleanUp {
+            withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+              AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(student.makeAuthToken())
+
+              // add "student" to billing project with owner role
+              Rawls.billing.addUserToBillingProject(projectName, student.email, BillingProjectRole.Owner)
+              register cleanUp Try(Rawls.billing.removeUserFromBillingProject(projectName, student.email, BillingProjectRole.Owner)).recover {
+                case _: RestException =>
```

Comment:
Log the exception (here and in other places in this file where we are following this pattern) so it's not just lost

Reply:
👍 Register a cleanUp here is redundant because fixtures `withGroup` and `withCleanBillingProject` should take care the cleanUp automatically.  I've removed this from all tests.